### PR TITLE
[sui-releases]: push sui binaries archives to s3

### PR DESCRIPTION
**Pull request by ebmifa**
## Description 
push sui binaries archives to s3

## Test Plan
https://github.com/MystenLabs/sui/actions/runs/6934597507

Upload to s3: https://s3.console.aws.amazon.com/s3/object/sui-releases?region=us-east-1&prefix=releases/sui-devnet-v1.14.0-ubuntu-x86_64.tgz

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -101,6 +108,7 @@ jobs:
           mv ./target/release/sui-indexer ./target/release/sui-indexer-${os_type}
           mv ./external-crates/move/target/release/move-analyzer ./external-crates/move/target/release/move-analyzer-${os_type}
           tar -cvzf ./target/release/sui-${{ env.sui_tag }}-${os_type}.tgz ./target/release/sui*-${os_type}* ./external-crates/move/target/release/move-analyzer-${os_type}
+          aws s3 cp ./target/release/sui-${{ env.sui_tag }}-${os_type}.tgz s3://sui-releases/releases/
```
Comment (johnjmartin) [inline .github/workflows/release.yml:111]:
```suggestion
          aws s3 cp ./target/release/sui-${{ env.sui_tag }}-${os_type}.tgz s3://sui-releases/releases/sui-${{ env.sui_tag }}-${os_type}.tgz
```

File: sui-node
Comment (johnjmartin) [inline sui-node:1]: accidental commit?

Comment (ebmifa) [reply sui-node:1]: yeah, left over garbadge

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -101,6 +108,7 @@ jobs:
           mv ./target/release/sui-indexer ./target/release/sui-indexer-${os_type}
           mv ./external-crates/move/target/release/move-analyzer ./external-crates/move/target/release/move-analyzer-${os_type}
           tar -cvzf ./target/release/sui-${{ env.sui_tag }}-${os_type}.tgz ./target/release/sui*-${os_type}* ./external-crates/move/target/release/move-analyzer-${os_type}
+          aws s3 cp ./target/release/sui-${{ env.sui_tag }}-${os_type}.tgz s3://sui-releases/releases/sui-${{ env.sui_tag }}-${os_type}.tgz
```
Comment (johnjmartin) [inline .github/workflows/release.yml:111]: We can probably get away with only pushing these on `mainnet` release

**Review by johnjmartin — APPROVED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (ebmifa) [reply .github/workflows/release.yml]: Added the check

**Review by johnjmartin — APPROVED** [review]

**Pull request merged**